elements/lisk-validator/src/keywords/data_type.ts

```diff
@@ -60,19 +69,27 @@ const compile = (
 	}
 
 	return (
-		_data: string,
+		_data: Buffer | bigint | string | number,
 		_dataPath?: string,
 		_parentData?: object,
 		_parentDataProperty?: string | number,
 		_rootData?: object,
-	): boolean =>
+	): boolean => {
+		if (value === 'boolean') return isBoolean(_data);
+		if (value === 'bytes') return isBytes(_data as Buffer);
+		if (value === 'string') return isString(value);
+		if (value === 'uint32') return isUInt32(_data);
+		if (value === 'sint32') return isSInt32(_data);
+		if (value === 'uint64') return isSInt64(_data);
+		if (value === 'uint64') return isUInt64(_data);
+
 		// Either "dataType" or "type" can be presented in schema
-		!typePropertyPresent;
+		return !typePropertyPresent;
```

Comment:
You can remove this line as well, earlier this line had no impact as we had fixed validation value. Now since we had some business logic in validation function this line is no more required.